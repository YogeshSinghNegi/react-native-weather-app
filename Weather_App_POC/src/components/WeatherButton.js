import * as AppColors from '../utils/AppColors';
import * as AppStrings from '../utils/AppStrings';

import { Button } from 'react-native-elements';
import React from 'react';
import { StyleSheet } from 'react-native';

const WeatherButton = (props) => {

    return (
        <Button 
          title = {AppStrings.GET_WEATHER} 
          buttonStyle = {Styles.button}
          onPress = {props.onPress}
        />
    );
};

const Styles = StyleSheet.create({
    button: {
        backgroundColor: AppColors.darkColor,
        color: AppColors.whiteColor,
        paddingHorizontal: 20,
        paddingVertical: 10,
        height: 45,
        borderRadius: 22.5,
        width: 200,
        fontSize: 18,
        textAlign: 'center',
      }
  });

export default WeatherButton;