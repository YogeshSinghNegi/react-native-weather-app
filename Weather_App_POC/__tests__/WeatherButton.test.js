import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import WeatherButton from '../src/components/WeatherButton';

configure({adapter: new Adapter()});

describe('<WeatherButton />', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<WeatherButton title= "test label"/>)
            expect(component).toMatchSnapshot()
        });
    });
});