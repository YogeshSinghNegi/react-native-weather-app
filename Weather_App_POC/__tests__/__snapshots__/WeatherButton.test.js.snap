// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherButton /> Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
